### PR TITLE
Create workspace from a node

### PR DESCRIPTION
**Pull request by Zulalm**
A menu is added to the node details page. This menu contains only one button to create new workspace from an existing node.
For now, this menu is visible to only authenticated users. I will change it to be visible to only contributors and reviewers, after merging this PR.

### CODE REVIEW

File: project/FrontEnd/collaborative_science_platform/lib/models/workspaces_page/entry.dart [hunk 1]
```diff
@@ -34,9 +34,9 @@ class Entry {
         isProofEntry: jsonString['is_proof_entry'],
         isTheoremEntry: jsonString['is_theorem_entry'],
         entryDate: DateTime.parse(jsonString['entry_date']),
-        entryId: jsonString['entry_id'],
-        entryNumber: jsonString['entry_number'],
-        index: jsonString['entry_index'],
+        entryId: jsonString['entry_id'] ?? 0,
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/models/workspaces_page/entry.dart:37]: It's better to set default id as -1

**Review by omerfaunal — CHANGES_REQUESTED** [review]

File: project/FrontEnd/collaborative_science_platform/lib/models/workspaces_page/entry.dart (hunk 1 above)
Comment (Zulalm) [reply project/FrontEnd/collaborative_science_platform/lib/models/workspaces_page/entry.dart:37]: I changed default values as -1.

**Review by laylaylo — CHANGES_REQUESTED** [review]
There are some interesting confusion between this PR and annotation PR, let's try to merge conflicts again.

**Review by laylaylo — APPROVED** [review]

**Comment by laylaylo** [comment]
Nice work!

**Review by omerfaunal — APPROVED** [review]
We talked together in the lab. Everything is OK now

**Pull request merged**